src/Illuminate/Support/Arr.php

```diff
@@ -111,7 +111,7 @@ public static function dot($array, $prepend = '')
         $results = [];
 
         foreach ($array as $key => $value) {
-            if (is_array($value) && ! empty($value)) {
+            if ($value !== []) {
```

Comment:
Are you sure that is_array() is unnecessary?
Values like 1, "foo", null and many others is not []. It will cause E_WARNING and it will work unpredictable.

Reply:
True that, this change is wrong: https://3v4l.org/egtb5

Reply:
I see. Thanks for feedback.

@carusogabriel In that case, the IsArray empty Rector can be dropped, right?

Reply:
Resolved https://github.com/rectorphp/rector/pull/779